### PR TITLE
Case 21227: don't enforce MOTION_TYPE_STATIC on mesh shapes

### PR DESCRIPTION
**Pull request by AndrewMeadows**
https://highfidelity.manuscript.com/f/cases/edit/21227/Near-grabbed-model-entities-with-exact-collision-do-not-update-for-observers

This PR fixes a bug where an Observer does not see the Grab effects on an ModelEntityItem that uses "Exact Collision" shape option.

### CODE REVIEW

**Comment by AndrewMeadows** [comment]
The reason I've added the **PRE-MERGE_TESTING_REQUIRED** label is because this change has potential consequences when moving other mesh shapes around.  We should have someone from QA check to see if the consequences are worse that the old ways in which moving mesh objects would introduce bad/incorrect physics.

**Comment by AndrewMeadows** [comment]
Jenkins, rebuild this please.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17331/HighFidelity-Beta-PR14932-165cfb0.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17331/HighFidelity-Beta-PR14932-165cfb0.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17331/HighFidelity-Beta-Interface-PR14932-165cfb0.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17331/HighFidelity-Beta-PR14932-165cfb0.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17331/HighFidelity-Beta-Interface-PR14932-165cfb0.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17360/HighFidelity-Beta-PR14932-2fdc9bc.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17360/HighFidelity-Beta-PR14932-2fdc9bc.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17360/HighFidelity-Beta-Interface-PR14932-2fdc9bc.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17360/HighFidelity-Beta-PR14932-2fdc9bc.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17360/HighFidelity-Beta-Interface-PR14932-2fdc9bc.exe

File: libraries/entities/src/EntityItem.cpp [hunk 1]
```diff
@@ -1969,23 +1959,19 @@ void EntityItem::setGravity(const glm::vec3& value) {
...
+            const float MAX_ANGULAR_SPEED = 9.0f * TWO_PI; // 1/10 rotation per step at 90Hz
+            if (speed < MIN_ANGULAR_SPEED) {
+                angularVelocity = ENTITY_ITEM_ZERO_VEC3;
+            } else if (speed > MAX_ANGULAR_SPEED) {
+                angularVelocity = (MAX_ANGULAR_SPEED / speed) * value;
+            } else {
+                angularVelocity = value;
             }
+            setLocalAngularVelocity(angularVelocity);
+            _flags |= Simulation::DIRTY_ANGULAR_VELOCITY;
```
Comment (NissimHadar) [inline libraries/entities/src/EntityItem.cpp]: Wouldn't it be neater to use a simple clamp function?

Comment (AndrewMeadows) [reply libraries/entities/src/EntityItem.cpp]: No.  When you clamp the components you change the direction of the vector.  This fancy clamping algorithm applies to the vector's length but does not rotate.

Comment (NissimHadar) [reply libraries/entities/src/EntityItem.cpp]: Oh, I see that now.

**Comment by hifi-gustavo** [comment]
Can someone from High Fidelity approve this PR?

**Comment by shanzzam** [comment]
removing the pre-merge flag requirement per sam and merging for integration testing

**Pull request merged**